includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -956,15 +968,22 @@ private function get_stylesheet_priority( DOMNode $node ) {
 				$priority += $print_priority_base;
 			}
 		} elseif ( $node instanceof DOMElement && 'style' === $node->nodeName ) {
-			$element_id = (string) $node->getAttribute( 'id' );
-			if ( 'admin-bar-inline-css' === $element_id ) {
-				$priority = $admin_bar_priority;
-			} elseif ( 'wp-custom-css' === $element_id ) {
-				// Additional CSS from Customizer.
-				$priority = 60;
-			} else {
-				// Other style elements, including from Recent Comments widget.
-				$priority = 70;
+			switch ( (string) $node->getAttribute( 'id' ) ) {
+				case 'twentytwenty-style-inline-css':
```

Comment:
This should be expanded to account any theme, not just Twenty Twenty. A quick way to do that would be:

```suggestion
				case get_template() . '-style-inline-css':
				case get_stylesheet() . '-style-inline-css':
```

Core themes starting with Twenty Twelve enqueue stylesheets like so:

```php
wp_enqueue_style( 'twentytwelve-style', get_stylesheet_uri(), array(), '20190507' );
```

A more robust way of determining the priority would be derive the underlying style handle from the ID and then look up the registered stylesheet to then determine if it is coming from a stylesheet.

```php
if ( $node->getAttribute( 'id' ) && 
     preg_match( '/^(?P<handle>.+)-inline-css$/', $node->getAttribute( 'id' ), $matches ) &&
     wp_style_is( $matches['handle'], 'registered' ) &&
     (
         0 === strpos( wp_styles()->registered[ $matches['handle'] ]->src, get_template_directory_uri() ) ||
         0 === strpos( wp_styles()->registered[ $matches['handle'] ]->src, get_stylesheet_directory_uri() )
     )
) {
    $priority = 2;
}
```

So I think it's pretty easily implementable, and it would resolve the below todo comment.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/3877/commits/e0ba933ed168ad2bd87d2d5e90bd99f7041b1f5c